spec/cases/with_exception_before_finish.rb

```diff
@@ -0,0 +1,20 @@
+require './spec/cases/helper'
+
+method = ENV.fetch('METHOD')
+in_worker_type = "in_#{ENV.fetch('WORKER_TYPE')}".to_sym
+
+begin
+  finish = lambda do |_item, _index, _result|
+    print " called"
+  end
+
+  Parallel.public_send(method, 1..10, in_worker_type => 4, finish: finish) do |x|
+    if x != 3
+      sleep 0.2
+      raise 'foo'
+    end
+    print x
+    x
+  end
+rescue
```

Comment:
maybe also raise something specific and then rescue only that ... to make sure we are not raising random errors